scripts/nbconverted/main_Pa_sim_enhance_AtoB.py

```diff
@@ -0,0 +1,283 @@
+    "size of gene set B": geneSetB_size,
+    "genes in set B": geneSetB,
+    "effect size of genes in set B": effect_size,
+    "threshold of A activation": thresholdA,
+}
+
+with open(log_file, 'w') as f:
+    for key, value in args_dict.items():
+        f.write('%s: %s\n' % (key, value))
+        
+# Output geneA only file using sample gene A selected
+geneA_file = os.path.join(base_dirs[0], analysis_name, geneSetA_pick + ".txt")
+geneA_only.to_csv(geneA_file, sep='\t', float_format="%.5g")
+
+
+# In[14]:
+
+
+get_ipython().run_cell_magic('time', '', '## Train compression methods using simulated data\n\n# Parameters to train nonlinear (VAE) compression method\nlearning_rate = 0.001\nbatch_size = 100\nepochs = 200\nkappa = 0.01\nintermediate_dim = 2775\nlatent_dim = 300\nepsilon_std = 1.0\nnum_PCs = latent_dim\n\nbase_dir = os.path.dirname(os.getcwd())\n\n# Train nonlinear (VAE)\nvae.tybalt_2layer_model(learning_rate, batch_size, epochs, kappa, intermediate_dim,\n                        latent_dim, epsilon_std, base_dir, analysis_name)\n# Train linear (PCA)\npca.pca_model(base_dir, analysis_name, num_PCs)')
+
+
+# In[15]:
+
+
+## Predict gene expression for each sample
+# Prediction based calculation described in interpolate.py
+
+# Define offset vectors in gene space
+data_dir = os.path.join(base_dirs[0], analysis_name)
+target_gene = geneSetA_pick
+percent_low = 5
+percent_high = 95
+
+def_offset.gene_space_offset(data_dir, target_gene, percent_low, percent_high)
+
+model_dir = os.path.join(base_dirs[2], analysis_name)
+encoded_dir = os.path.join(base_dirs[1], analysis_name)
+
+# Define offset vectors for VAE latent space
+def_offset.vae_latent_space_offset(data_dir, model_dir, encoded_dir, latent_dim, target_gene, percent_low,
+                                   percent_high)
+# Define offset vectors for PCA latent space
+def_offset.pca_latent_space_offset(data_dir, model_dir, encoded_dir, target_gene, percent_low, percent_high)
+
+
+# Predict gene expression using offset in gene space and latent space
+out_dir = os.path.join(base_dirs[3], analysis_name)
+
+interpolate.interpolate_in_gene_space(data_dir, target_gene, out_dir, percent_low, percent_high)
+interpolate.interpolate_in_vae_latent_space(data_dir, model_dir, encoded_dir, latent_dim, 
+                                            target_gene, out_dir, percent_low, percent_high)
+interpolate.interpolate_in_pca_latent_space(data_dir, model_dir, encoded_dir, target_gene, 
+                                            out_dir, percent_low, percent_high)
+
+
+# In[18]:
+
+
+## Visualize prediction performance
+
+# True if the x-axis of the plot uses the sample index
+# False if the x-asix of the plot uses the gene expression of the target gene
+by_sample_ind = False
+
+# Plot prediction per sample along gradient of gene A expression
+viz_dir = os.path.join(base_dirs[5], analysis_name)
+plot.plot_corr_gradient(out_dir, viz_dir, target_gene, by_sample_ind)
```

Comment:
assuming this somehow draws information from the `interpolate` methods above

Reply:
Yes that's correct.  The interpolate will return the correlation scores per sample that are then plotted.  I will also add a comment for this as well.

Reply:
Yes exactly